docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -642,6 +642,146 @@ class MyGallery extends React.Component {
 export default withInjector(MyGallery);
 ```
 
+### Using injector to customise redux state data
+
+It is recommended to be familiar with the concepts of Immutability and Redux before starting this.
+
+The examples use `Spread in object literals` which is at this moment in Stage 3 Proposal. If you're more comfortable with using
+ the `Object.assign()` API that shouldn't present any problems and should work the same.
+
+To start customising, you'll need to transform an existing registered reducer.
+
+```js
+Injector.transform('customisationName', (update) => {
+  update.reducer('assetAdmin', MyReducerTransformer);
+});
+```
+
+Alternatively, you can also register your own reducer to injector, particularly if you're building your own module.
+
+```js
+Injector.reducer.register('myCustom', MyCustomerReducer);
+```
+
+#### Structuring a transformer
+
+We utilise currying to supply utilities which your transformer may require to handle the transformation.
+- `originalReducer` - reducer callback which the transformer is customising, this will need to be called in most cases. This will also callback other transformations down the chain of execution, not calling this will break the chain.
+- `globalState` - state of the global redux store, there may be data outside the current scope in the reducer which you may need to help determine the transformation.
+- `state` - current state of the current scope, this is what should be used to form the new State.
+- `type` - the action to fire, like in any reducer in redux, this helps determine if your transformer should do anything
+- `payload` - the new data sent with the action to mutate the redux store.
+
+```js
+const MyReducerTransformer = (originalReducer) => (globalState) => (state, { type, payload }) => {
+  switch (type) {
+    case 'MY_ACTION': {
+      // recommended to call and return the originalReducer with the payload changed by the transformer
+      /* return action to call here; */
```

Comment:
It's hard to tell the difference between the first and second `case` here. Maybe rename to `CUSTOM_ACTION` and `EXISTING_ACTION`?